editor/src/viewport_tools/tool.rs

```diff
@@ -65,6 +66,53 @@ impl ToolData {
 	}
 }
 
+impl PropertyHolder for ToolData {
+	fn properties(&self) -> WidgetLayout {
+		let tool_groups_layout = ToolType::list_tools_in_groups()
+			.iter()
+			.flat_map(|group| {
+				let separator = std::iter::once(WidgetHolder::new(Widget::Separator(Separator {
+					direction: SeparatorDirection::Vertical,
+					separator_type: SeparatorType::Section,
+				})));
+				let buttons = group.iter().map(|tool_type| {
+					WidgetHolder::new(Widget::IconButton(IconButton {
+						icon: tool_type.icon_name(),
+						size: 32,
+						tooltip: tool_type.tooltip(),
+						active: self.active_tool_type == *tool_type,
+						on_update: WidgetCallback::new(|_| {
+							if !tool_type.tooltip().contains("Coming Soon") {
+								ToolMessage::ActivateTool { tool_type: *tool_type }.into()
+							} else {
+								DialogMessage::RequestComingSoonDialog { issue: None }.into()
+							}
+						}),
+						..Default::default()
+					}))
+				});
+				separator.chain(buttons)
+			})
+			// Skip the initial separator
+			.skip(1)
+			.collect();
+
+		WidgetLayout {
+			layout: vec![LayoutRow::Column { widgets: tool_groups_layout }],
+		}
+	}
+
+	fn register_properties(&self, responses: &mut VecDeque<Message>, layout_target: LayoutTarget) {
+		responses.push_back(
+			LayoutMessage::SendLayout {
+				layout: self.properties(),
+				layout_target,
+			}
+			.into(),
+		)
+	}
```

Comment:
This doesn't need to be implemented manually (as it is the same as the default implementation)